src/document-converter.ts

```diff
@@ -248,15 +271,29 @@ export class DocumentConverter {
 
     const combinedToplevelStatements = [];
     let prevScriptNode: parse5.ASTNode|undefined = undefined;
-    for (const script of this.document.getFeatures({kind: 'js-document'})) {
+    const allFeatures = this.document.getFeatures();
+    const relevantScripts = Array.from(allFeatures).filter((f) => {
+      if (f.kinds.has('js-document')) {
+        return true;
+      }
+      if (f.kinds.has('html-script')) {
+        return this.isInternalNonModuleImport(f as Import);
+      }
+      return false;
+    });
+
+    for (const script of relevantScripts) {
+      const scriptDocument = script.kinds.has('js-document') ?
```

Comment:
For the external scripts that don't have the same path as this document, we need to return a result that marks them for deletion.

Reply:
just clarifying: you mean external to the current file, and not to the current package, correct?

Reply:
yes :)

basically, we should delete the files we inline and don't overwrite (or delete all inlined files as long as the order of writing results works out)

Reply:
got it, just making sure I understood you